x-pack/plugin/watcher/src/main/java/org/elasticsearch/xpack/watcher/actions/index/IndexAction.java

```diff
@@ -143,6 +156,13 @@ public static IndexAction parse(String watchId, String actionId, XContentParser
             } else if (token == XContentParser.Token.VALUE_STRING) {
                 if (Field.DOC_ID.match(currentFieldName, parser.getDeprecationHandler())) {
                     docId = parser.text();
+                } else if (Field.OP_TYPE.match(currentFieldName, parser.getDeprecationHandler())) {
+                    try {
+                        opType = DocWriteRequest.OpType.fromString(parser.text());
```

Comment:
maybe validate that only create or index op type can be used?

Reply:
Thanks, @martijnvg. What versions do you think we should target for this?

Reply:
8.0.0 and 7.11?